url5a01f2c1d9_core9_core_tgz-pJ8-io_core9_core_boot_BootstrapFrameworkJ8/src/io/core9/core/boot/BootstrapPluginsImpl.java

```diff
@@ -0,0 +1,75 @@
+package io.core9.core.boot;
+
+import io.core9.core.Config;
+import io.core9.core.PluginRegistry;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Collection;
+
+import net.xeoh.plugins.base.Plugin;
+import net.xeoh.plugins.base.PluginManager;
+import net.xeoh.plugins.base.impl.spawning.handler.InjectHandler;
+import net.xeoh.plugins.base.options.addpluginsfrom.OptionReportAfter;
+
+/**
+ * Bootstraps the plugins in the /plugins folder
+ * 
+ * @author mark
+ * 
+ */
+public class BootstrapPluginsImpl implements BootstrapPlugins {
+
+	private Collection<Plugin> plugins;
+	private PluginRegistry registry;
+
+	@Override
+	public void bootstrap() {
+
+		Config config = Config.getInstance();
+		
+		String[] pluginDirs = config.getPluginDirs();
+		for (int i = 0; i < pluginDirs.length; i++) {
+			registry.getManager().addPluginsFrom(new File(pluginDirs[i]).toURI(), new OptionReportAfter()); 	
+		}
+		try {
+			registry.getManager().addPluginsFrom(new URI("classpath://*"), new OptionReportAfter());
+		} catch (Exception e) {	
+			e.printStackTrace();
+		}
+
+		
+		plugins = registry.getManager().getPluginRegistry().getAllPlugins();
+				
+		for (Plugin plugin : plugins) {
+			registry.registerPlugin(plugin.getClass(), plugin);
+			PluginManager pluginManager = registry.getManager();
+			try {
+				// this is only possible when all plugins are loaded
+	            new InjectHandler(pluginManager).init(plugin);
+            } catch (Exception e) {
+	            // TODO Auto-generated catch block
+	            e.printStackTrace();
+            }
+		}
+	}
+
+	@Override
+	public void setRegistry(PluginRegistry registry) {
+		this.registry = registry;
+	}
+	
+	public static void addPath(String s) throws Exception {
+	    File f = new File(s);
+	    URL u = f.toURI().toURL();
+	    System.out.println("Adding files from: " + u.toString());
+	    URLClassLoader urlClassLoader = (URLClassLoader) ClassLoader.getSystemClassLoader();
+	    Class<URLClassLoader> urlClass = URLClassLoader.class;
+	    Method method = urlClass.getDeclaredMethod("addURL", new Class[]{URL.class});
+	    method.setAccessible(true);
+	    method.invoke(urlClassLoader, new Object[]{u});
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)